File: typescript/src/shapes/circle.ts
interface Circle {
    // defined private properties
    radius: number,
    computeArea: () => number
}

function newCircle(radius: number): Shape {
    return {
        computeArea: function (): number {
            return Math.PI * radius * radius;
        }
    };
}

export { Circle, newCircle }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Circle.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Circle implements Shape {
    // change visibility of radius from private to protected
    private double radius;

    public Circle(double radius) {
        this.radius = radius;
    }

    public double getArea() {
        return Math.PI * radius * radius;
    }
}
